src/main/java/com/google/cloud/genomics/dataflow/utils/GenomicsDatasetOptions.java

```diff
@@ -37,18 +38,18 @@
 
     // TODO: If needed, add getReadRequests method
     public static List<SearchVariantsRequest> getVariantRequests(GenomicsDatasetOptions options,
-        GenomicsFactory.OfflineAuth auth, boolean excludeXY)
-        throws IOException, GeneralSecurityException {
+        GenomicsFactory.OfflineAuth auth, boolean excludeXY) throws IOException,
+        GeneralSecurityException {
       String datasetId = options.getDatasetId();
       Genomics genomics = auth.getGenomics(auth.getDefaultFactory());
 
-      Iterable<Contig> contigs = options.isAllContigs()
-          ? Contig.getContigsInVariantSet(genomics, datasetId, excludeXY)
-          : Contig.parseContigsFromCommandLine(options.getReferences());
+      Iterable<Contig> contigs =
+          options.isAllContigs() ? Contig.getContigsInVariantSet(genomics, datasetId, excludeXY)
+              : Contig.parseContigsFromCommandLine(options.getReferences());
 
       List<SearchVariantsRequest> requests = Lists.newArrayList();
       for (Contig contig : contigs) {
-        for (Contig shard : contig.getShards()) {
+        for (Contig shard : contig.getShards(options.getBasesPerShard())) {
```

Comment:
You might consider *not* doing this, since it will add more lines to your code (at least it will in Java 7, in Java 8 it won't), but this computation looks like a pretty simple stream accumulation:

    FluentIterable
        .from(contigs)
        .transformAndConcat(
            new Function<Contig, Iterable<Contig>>() {
              @Override public Iterable<Contig> apply(Contig contig) {
                return contig.getShards(options.getBasesPerShard());
              }
            })
        .transform(
            new Function<Contig, SearchVariantsRequest>() {
              @Override public SearchVariantsRequest apply(Contig shard) {
                return shard.getVariantsRequest(datasetId);
              }
            })
        .toList();

FYI, in Java 8, it would simply be:

    contigs
        .stream()
        .flatMap(contig -> contig.getShards(options.getBasesPerShard()))
        .map(shard -> shard.getVariantsRequest(datasetId))
        .collect(Collectors.toList());